### ISSUE TITLE
False positive: Defining a new constructor causes a type to longer be understood as a type

### ISSUE THREAD

**Issue by dzhang314** [julia-vscode/StaticLint.jl#395]
Consider the following Julia code:

```
module M

struct MyNumber
    sign::Bool
    exponent::Int
    mantissa::Int
end

function BigFloat(::MyNumber)
    ...
end

function cube_root(x::BigFloat)
    ...
end

end # module M
```

This yields blue squiggles under the definition of function `cube_root(x::BigFloat)` with the following message:

```
A non-DataType has been used in a type declaration statement. Julia(InvalidTypeDeclaration)
```

Here, the presence of a new constructor `BigFloat(::MyNumber)` has caused StaticLint to stop regarding `BigFloat` as a type. This pattern occurs a few times in my package [**MultiFloats.jl**](https://github.com/dzhang314/MultiFloats.jl).

**Comment by pfitzseb** [comment]
The reproducer as written is actually invalid. https://github.com/julia-vscode/StaticLint.jl/pull/431 adds some regression tests for the valid cases (qualified/imported method def)
